packages/core/src/rum/DdRum.ts

```diff
@@ -230,24 +230,20 @@ class DdRumWrapper implements DdRumType {
         );
     };
 
-    generateUUID = (type: TracingIdType): string => {
-        switch (type) {
-            case TracingIdType.trace:
-                return TracingIdentifier.createTraceId().toString(
-                    TracingIdFormat.paddedHex
-                );
-            case TracingIdType.span:
-                return TracingIdentifier.createSpanId().toString(
-                    TracingIdFormat.decimal
-                );
-            default:
-                console.warn(
-                    `Unsupported tracing ID type '${type}' for generateUUID. Falling back to 64 bit Span ID.`
-                );
-                return TracingIdentifier.createSpanId().toString(
-                    TracingIdFormat.decimal
-                );
-        }
+    /**
+     * Generates a UUID based on the provided tracing type and propagator.
+     * @param tracingIdType The tracing ID type (trace/span).
+     * @param propagatorType The context propagation format.
+     * @returns Object containing the formatted UUID for resource and context propagation.
+     */
+    public generateUUID = (tracingIdType: TracingIdType): FormattedUUID => {
```

Comment:
I think using wording `UUID` here may create a confusion, because `UUID` is a fixed-length 128 bit identifier https://en.wikipedia.org/wiki/Universally_unique_identifier, while what is returned by this method is different.

Also maybe it is not worth to hide both span ID and trace ID generation behind the single method. I would suggest to have dedicated methods `generateTraceId` and `generateSpanId` which return the object having the methods to convert them to string in different formats: fixed-length hex, var-length hex, decimal.

This is the approach taken in iOS SDK, Android SDK.

But this is just a suggestion from my side anyway.